icu4c/source/i18n/units_router.cpp

```diff
@@ -43,8 +43,24 @@ Precision UnitsRouter::parseSkeletonToPrecision(icu::UnicodeString precisionSkel
     return result;
 }
 
+UnitsRouter::UnitsRouter(const char *inputUnitIdentifer, StringPiece region, StringPiece usage,
```

Comment:
Also: since MeasureUnit::forIdentifier accepts a StringPiece, it is more efficient to have UnitsRouter also accept a StringPiece: it saves having to find the char*'s length multiple times (which is O(N) - reading the whole char* looking for a null).

At call sites: CharString already knows its own length, and has a toStringPiece() method.

Reply:
+1 on taking a StringPiece rather than `const char*`

Reply:
done